### PR TITLE
Add Size_of_largest_BST_in_binary_tree.md

### LINKED ISSUE #1851: [Custom]: <ADD A PROBLEM> Size of largest BST in binary tree
### Feature Name

Size of largest BST in binary tree

### Feature Description

Size of largest BST in binary tree , we have to find the largest subtree in the binary tree which is also BST.

### Motivation

An intuitive problem to solve from interviews point of view , will help to understand a different way to traverse in the binary tree.

### Implementation Suggestions (Optional)

- We will traverse the tree in a bottom-up manner.
- For every traversed node, we return the maximum and minimum values in the subtree rooted at that node.
- The idea is to determine whether the subtree rooted at each node is a Binary Search Tree (BST).
- If any node follows the properties of a BST (largest value in left Subtree is smaller than the root value and smallest value in the right Subtree is greater than the root value)
- For subtree having the maximum size, we will update the size of the largest BST.

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [ ] UI/UX Design
- [ ] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by Riddhi12349**
### Description
Size of largest BST in binary tree by recursively calling for left and right subtrees and return the maximum of two calls. 

Fixes #1851 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/DSA-Problem-Solution/Size_of_largest_BST_in_binary_tree.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+id: size-of-largest-bst-in-binary-tree
+title: Size of Largest BST in Binary Tree
+sidebar_label: GFG
+tags: [GFG , Binary Tree , BST]
+description: A Tree is BST if following is true for every root node x.
+
+The largest value in left subtree (of x) is smaller than value of x.
+The smallest value in right subtree (of x) is greater than value of x.
+
+We will traverse the tree in a bottom-up manner. For every traversed node, we return the maximum and minimum values in the subtree rooted at that node. The idea is to determine whether the subtree rooted at each node is a Binary Search Tree (BST) (for subtree to be bst conditions mentioned above). If any node follows the properties of a BST and has the maximum size, we update the size of the largest BST.
```
Comment (ajay-dhangar) [inline docs/DSA-Problem-Solution/Size_of_largest_BST_in_binary_tree.md]: delete line `7` to `11`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Riddhi12349** [comment]
done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**